### PR TITLE
feat: add custom asset sorting

### LINKED ISSUE #4761: [Task] Assets get reordered when loging in
This appears to be a walletrs issue, because account.meta.nativeTokens is in different order after syncing

### PR DESCRIPTION
**Pull request by MarkNerdi**
## Summary

This PR includes the new OrderFilterUnit which allows to order the assets

## Changelog

```
- sort assets by different criteria
- refactor some asset logic
```

## Relevant Issues

Closes #4761

### Platforms

> Please select any platforms where your changes have been tested.

- __Desktop__
  - [ ] MacOS
  - [ ] Linux
  - [ ] Windows
- __Mobile__
  - [ ] iOS
  - [ ] Android

### Instructions

> Please describe the specific instructions, configurations, and/or test cases necessary to __test and verify__ that your changes work as intended.

...

## Checklist

> Please tick all of the following boxes that are relevant to your changes.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or modified tests that prove my changes work as intended
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have verified that my latest changes pass CI workflows for testing and linting
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: packages/shared/lib/core/wallet/utils/sortAssets.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+function sortByName(asset1: IAsset, asset2: IAsset, asc: boolean): number {
+    return asset1.metadata.name.toLowerCase() > asset2.metadata.name.toLowerCase() ? (asc ? -1 : 1) : asc ? 1 : -1
+}
+
+function sortByAmount(asset1: IAsset, asset2: IAsset, asc: boolean): number {
+    return asset1.balance.total > asset2.balance.total ? (asc ? 1 : -1) : asc ? 0.1 : 1
+}
+
+function sortByDate(asset1: IAsset, asset2: IAsset, asc: boolean): number {
+    return asset1.balance.total > asset2.balance.total ? (asc ? 1 : -1) : asc ? 0.1 : 1
```
Comment (MarkNerdi) [inline packages/shared/lib/core/wallet/utils/sortAssets.ts:38]: we need to find a way to determine date

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Getting an error when logging out on this branch

**Review by nicole-obrien — CHANGES_REQUESTED** [review]
Balance is not changing in the asset list in this example I sent all the BTC balance

Screenshot 2022-10-12 at 10 59 38

**Review by nicole-obrien — APPROVED** [review]
LGTM!

**Pull request merged**